util/random/rand.go

```diff
@@ -76,3 +81,59 @@ func (r *randstream) XORKeyStream(dst, src []byte) {
 func New() cipher.Stream {
 	return &randstream{}
 }
+
+// READER_BYTES is how many bytes we expect from each source
+const READER_BYTES = 32
+
+// MIXEDRAND_HASH is the hash used in XORKeyStream to mix all entropy sources together.
+// sha512 is chosen as its digest is 64 bytes long, matching the length of blake2xb's seed.
+var MIXEDRAND_HASH = sha512.New
+
+type mixedrandstream struct {
+	Readers []io.Reader
+}
+
+func (r *mixedrandstream) XORKeyStream(dst, src []byte) {
+
+	l := len(dst)
+	if len(src) != l {
+		panic("XORKeyStream: mismatched buffer lengths")
+	}
+
+// try to read 32 bytes from all readers and write them in a buffer
+	var b bytes.Buffer
+	var nerr int
+	for _, reader := range r.Readers {
+		buff := make([]byte, READER_BYTES)
+		n, err := io.ReadFull(reader, buff)
+		if err != nil {
+			nerr++
+		}
+		b.Write(buff[:n])
+	}
+
+	// we are ok with few sources being insecure (i.e., providing less than
+	// READER_BYTES bytes), but not all of them
+	if nerr == len(r.Readers) {
+		panic("all readers failed")
+	}
+
+	// create the XOF output, with hash of collected data as seed
+	h := MIXEDRAND_HASH()
+	h.Write(b.Bytes())
+	seed := h.Sum(nil)
+	blake2 := blake2xb.New(seed)
+	blake2.XORKeyStream(dst, src)
+}
+
+// NewMixedStream returns a new cipher.Stream that gets random data from the given
+// readers. If no reader was provided, Go's crypto/rand package is used.
+// In order to use every source, it tries to read 32 bytes from each, and mix it
+// all together into a string of valid length via hashing. This string is then
+// used as a seed for blake2, which is used to perform the final XOR.
+func NewMixedStream(readers ...io.Reader) cipher.Stream {
```

Comment:
I'm still more in favor of keeping the same API as before `New()` so we don't introduce a new struct and API for having the same behavior basically. -> what I'm saying is I dont think we need to have another struct for that feature.

But Jeff's opinion should weight more in at this point cc/ @jeffallen

Reply:
This would be in order to avoid having something like 
`func New(readers ...io.Reader) cipher.Stream {
	if len(readers) == 0 {
		return &randstream{}
	} else {
	        return &mixedrandstream{readers}
        }
}` ?

Reply:
We normally prefer simpler APIs, even if that requires a more complex internal implementation. So yes, I would like to see `func New(readers ...io.Reader)`, that's a nice backwards compatible, and easy to use API.

Reply:
And do you want to keep the two structs separated or should I merge them ?

Reply:
In an ideal world, I'd merge them. Problem is that I think if we merge them, since we change the output of the regular `random.New()` struct, a lot of tests / examples  in kyber might fail now. 
If you can try that, that'd be awesome. If it doesn't fail, then let's merge. If it does, then let's not merge.

Reply:
travis is happy when I merge the two :) so should we keep it that way ?